frontend/src/core/i18n/__tests__/locale-provider.test.tsx

```diff
@@ -0,0 +1,154 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+
+import { cleanup, render } from "@testing-library/react";
+import { createStore, Provider } from "jotai";
+import { afterEach, describe, expect, it, vi } from "vitest";
+import { userConfigAtom } from "@/core/config/config";
+import { parseUserConfig } from "@/core/config/config-schema";
+import { LocaleProvider } from "../locale-provider";
+
+// Mock react-aria-components I18nProvider
+vi.mock("react-aria-components", () => ({
+  I18nProvider: ({
+    children,
+    locale,
+  }: {
+    children: React.ReactNode;
+    locale?: string;
+  }) => (
+    <div data-testid="i18n-provider" data-locale={locale}>
+      {children}
+    </div>
+  ),
+}));
+
+describe("LocaleProvider", () => {
+  afterEach(() => {
+    cleanup();
+  });
+
+  it("should render I18nProvider without locale when locale is null", () => {
+    const store = createStore();
+    const config = parseUserConfig({ display: { locale: null } });
+    store.set(userConfigAtom, config);
+
+    const { getByTestId } = render(
+      <Provider store={store}>
+        <LocaleProvider>
+          <div>Test content</div>
+        </LocaleProvider>
+      </Provider>,
+    );
+
+    const i18nProvider = getByTestId("i18n-provider");
+    expect(i18nProvider).toBeInTheDocument();
+    expect(i18nProvider.dataset.locale).toBe(undefined);
+    expect(i18nProvider).toHaveTextContent("Test content");
+  });
+
+  it("should render I18nProvider without locale when locale is undefined", () => {
+    const store = createStore();
+    const config = parseUserConfig({ display: { locale: undefined } });
+    store.set(userConfigAtom, config);
+
+    const { getByTestId } = render(
+      <Provider store={store}>
+        <LocaleProvider>
+          <div>Test content</div>
+        </LocaleProvider>
+      </Provider>,
+    );
+
+    const i18nProvider = getByTestId("i18n-provider");
+    expect(i18nProvider).toBeInTheDocument();
+    expect(i18nProvider.dataset.locale).toBe(undefined);
+    expect(i18nProvider).toHaveTextContent("Test content");
+  });
+
+  it("should render I18nProvider with locale when locale is provided", () => {
+    const store = createStore();
+    const testLocale = "es-ES";
+    const config = parseUserConfig({ display: { locale: testLocale } });
+    store.set(userConfigAtom, config);
+
+    const { getByTestId } = render(
+      <Provider store={store}>
+        <LocaleProvider>
+          <div>Test content</div>
+        </LocaleProvider>
+      </Provider>,
+    );
+
+    const i18nProvider = getByTestId("i18n-provider");
+    expect(i18nProvider).toBeInTheDocument();
+    expect(i18nProvider.dataset.locale).toBe(testLocale);
+    expect(i18nProvider).toHaveTextContent("Test content");
+  });
+
+  it("should render I18nProvider with different locale values", () => {
+    const testCases = ["en-US", "fr-FR", "de-DE", "ja-JP"];
```

Comment:
Do we want some test for behavior of an invalid locale config?

Reply:
im not running this through react-aria specifically (its mocked). i can try it out though